### DISCUSSION TITLE
Create provider to the separate TabPanes

### DISCUSSION THREAD

**Discussion by devjcob** [Textualize/textual#3809] [Q&A]
Hello, I have a `TabbedContent` with three `TabPanes`. I want to use the `set_interval` method to create a separate data refresh for each managed by `context manager`. The problem is that when I run `TabbedContent`, all the intervals also run. Is there any way to control the interval refresh depending on which `TabPane` is now active? An example of one of the `TabPane` is below.
```
from textual.app import App, ComposeResult
from textual.widgets import TabbedContent, TabPane, Static
from textual.containers import Container
from textual.widget import Widget


class DataProvider(Widget):
    def __init__(self):
        super().__init__()
        self.set_interval(1.5, self.example_callback)

    def example_callback(self) -> None:
        print("!")


class Body(Container):
    DEFAULT_CSS = """
    Body {
        height: 1fr;
        width: 1fr;
    }
    """


class MyTabPane(TabPane):
    def compose(self) -> ComposeResult:
        with Body(), DataProvider():
            yield Static("!!")


class MyApp(App):
    def compose(self) -> ComposeResult:
        with TabbedContent():
            yield MyTabPane("first tab")


if __name__ == "__main__":
    app = MyApp()
    app.run()
    
```

**Comment by willmcgugan** [comment]
The `set_interval` method returns a timer. Which you can pause and resume (https://textual.textualize.io/api/timer/#textual.timer.Timer).

You could catch the TabActivate event to pause and resume timers https://textual.textualize.io/widgets/tabbed_content/#messages

**Reply by devjcob** [reply]
Thanks for the advice, I wanted to do this using the `context manager`, but that will probably be the way to go. But perhaps you have a better way of refreshing and reusing data in a few widgets in the background in textual ?

**Reply by devjcob** [reply]
I don't like the way these providers deliver data to the widgets, as I showed in the example, and I thought you might have a better way, but I didn't know about it.
